packages/utils-dom/src/htmlelement.ts

```diff
@@ -233,7 +235,11 @@ export class HTMLElement extends Node {
      * https://developer.mozilla.org/en-US/docs/Web/API/Element/matches
      */
     public matches(selector: string): boolean {
-        return this.ownerDocument.querySelectorAll(selector).includes(this);
+        if (!CACHED_CSS_SELECTORS.has(selector)) {
+            CACHED_CSS_SELECTORS.set(selector, compile(selector));
```

Comment:
Can we move the compilation step into the refactored module as well? Then we can just expose a single `getCompiledSelector` method and not have to deal with multiple consumers messing directly with the `CACHED_CSS_SELECTORS` collection.

Reply:
Thanks, done!